tests/config/test_config.py

```diff
@@ -11,22 +12,36 @@
 import pytest
 
 from poetry.config.config import Config
+from poetry.config.config import boolean_normalizer
+from poetry.config.config import int_normalizer
 
 
 if TYPE_CHECKING:
     from pathlib import Path
 
 
-def get_boolean_options(config: Optional[Dict[str, Any]] = None) -> str:
+def flatten(config: Optional[Dict[str, Any]], parent_key: str = "", sep="."):
+    items = []
+    for k, v in config.items():
+        new_key = parent_key + sep + k if parent_key else k
+        if isinstance(v, dict):
+            items.extend(flatten(v, new_key, sep=sep).items())
+        else:
+            items.append((new_key, v))
+    return dict(items)
+
+
+def get_options_based_on_normalizer(
+    normalizer: Callable, config: Optional[Config] = None
+) -> str:
     if config is None:
-        config = Config.default_config
+        config = Config()
 
-    for k, v in config.items():
-        if isinstance(v, bool) or v is None:
+    flattened_config = flatten(config.default_config)
```

Comment:
Doesn't this make passing in a `Config` meaningless as we'll always use the static default? Feels like the original was more correct.

Reply:
Yes you're right passing a `Config` would be meaningless, but also in the original implementation it was confusing in my eyes. The problem why it needed to change is that the check for `None` does not indicate a `bool` config variable anymore.
So either the config names need to be hard-coded again or the normalizer can be used.

Reply:
I think it's fine to hardcode to always use the default config (`Config.default_config`) and we can modify this if another test case needs to pass one in later.